src/client/src/orangehrmPimPlugin/pages/employee/SaveEmployee.vue

```diff
@@ -205,13 +220,17 @@ export default {
         firstName: [required, shouldNotExceedCharLength(30)],
         middleName: [shouldNotExceedCharLength(30)],
         lastName: [required, shouldNotExceedCharLength(30)],
-        employeeId: [shouldNotExceedCharLength(10)],
+        employeeId: [
+          this.employeeIdUniqueValidation,
+          shouldNotExceedCharLength(10),
+        ],
         empPicture: [
           maxFileSize(1024 * 1024),
           validFileTypes(this.allowedImageTypes),
         ],
         username: [
           required,
+          this.usernameUniqueValidation,
           shouldNotLessThanCharLength(5),
           shouldNotExceedCharLength(40),
```

Comment:
```suggestion
          required,
          shouldNotLessThanCharLength(5),
          shouldNotExceedCharLength(40),
          this.usernameUniqueValidation,
```
:thinking:  I feel this should be the order. first check client side validation first and then only we should pass to the BE. what do you think

Reply:
I had it like that initially but due to the debounce the length validation would not work properly for certain cases